tools/first_observation_experiment.rb

```diff
@@ -13,13 +15,135 @@
 
 where [options] are:
 EOS
-  opt :user_data_path, "Path to the user data file.", type: :string, short: "-t"
+  opt :cohort_data_path, "Path to the cohort data file.", type: :string, short: "-t"
+end
+
+def get_obs( day, users )
+  filter = build_filter( day.beginning_of_day, day.end_of_day, users )
+  agg = build_agg( users )
+
+  obs_data = fetch_obs_data( filter, agg )
+
+  build_result_hash( obs_data )
 end
 
-# Validate user_data_path option
-unless OPTS.user_data_path
-  puts "You must specify a user data path"
-  exit( 1 )
+def get_captives( day, users )
+  filter = build_filter( day.beginning_of_day, day.end_of_day, users, captives: true )
+  agg = build_agg( users )
+
+  obs_data = fetch_obs_data( filter, agg )
+
+  obs_data.map {| a | a["key"] }
+end
+
+def get_obs_span( start_date, end_date, users )
+  filter = build_filter( start_date.beginning_of_day, end_date.end_of_day, users )
+  agg = build_agg( users )
+
+  obs_data = fetch_obs_data( filter, agg )
+
+  build_result_hash( obs_data )
+end
+
+def get_captives_span( start_date, end_date, users )
+  filter = build_filter( start_date.beginning_of_day, end_date.end_of_day, users, captives: true )
+  agg = build_agg( users )
+
+  obs_data = fetch_obs_data( filter, agg )
+
+  obs_data.map {| a | a["key"] }
+end
+
+def build_filter( start_time, end_time, users, captives: false )
+  filter = [
+    {
+      range: {
+        created_at: {
+          gte: start_time,
+          lte: end_time
+        }
+      }
+    },
+    {
+      terms: {
+        "user.id": users
+      }
+    }
+  ]
+
+  if captives
+    filter += [
+      { exists: { field: "observed_on" } },
+      { exists: { field: "location" } },
+      {
+        bool: {
+          should: [
+            { term: { captive: true } },
+            { term: { "taxon.id": 43_584 } },
```

Comment:
would be good to look these taxa up dynamically rather than hard code an ID. Maybe we'd want a Taxon::HUMAN constant similar to the Taxon::LIFE constant we already have, or at least get the ID from a query? This also looks like it would return observations that are captive OR genus Homo OR species Homo sapiens. Just want to make sure that's what you intended